File: backend-api/src/main/java/com/smartenergy/backendapi/model/Incidencia.java
package com.smartenergy.backendapi.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "Incidencia")
public class Incidencia extends BaseEntity{
    private String descripcion;
    private LocalDateTime fechaHora;
    private String tipoIncidencia;
    @Enumerated(EnumType.STRING)
    private EstadoAlerta estado;
    @ManyToOne
    @JoinColumn(name = "aula_id")
    private Aula aula;
}

File: mobile-app/app/src/main/java/com/example/smartenergy/model/Incidencia.kt
package com.example.smartenergy.model

import java.time.LocalDateTime
import java.util.UUID


data class Incidencia(
    val id: String? = null,
    val descripcion: String,
    val fechaHora: LocalDateTime = LocalDateTime.now(),
    val tipoIncidencia: String,
    val aula: Aula? = null,
    val estado: EstadoAlerta = EstadoAlerta.PENDIENTE
) {
    val tipo: String get() = tipoIncidencia
    val fecha: String get() = fechaHora.format(java.time.format.DateTimeFormatter.ofPattern("dd/MM/yyyy"))
}
